src/Refactoring-Core/RBRemoveMethodRefactoring.class.st

```diff
@@ -7,42 +7,34 @@ Class {
 	#name : #RBRemoveMethodRefactoring,
 	#superclass : #RBMethodRefactoring,
 	#instVars : [
-		'selectors'
+		'selector',
+		'transformation'
 	],
 	#category : #'Refactoring-Core-Refactorings'
 }
 
 { #category : #'instance creation' }
-RBRemoveMethodRefactoring class >> model: aRBNamespace removeMethods: selectorCollection from: aClass [
+RBRemoveMethodRefactoring class >> model: aRBNamespace selector: aSelector from: aClass [
 
 	^ self new
 		  model: aRBNamespace;
-		  removeMethods: selectorCollection from: aClass;
+		  selector: aSelector from: aClass;
 		  yourself
 ]
 
 { #category : #'instance creation' }
-RBRemoveMethodRefactoring class >> removeMethods: selectorCollection from: aClass [
-	^ self new
-		removeMethods: selectorCollection from: aClass
-]
-
-{ #category : #preconditions }
-RBRemoveMethodRefactoring >> applicabilityPreconditions [
+RBRemoveMethodRefactoring class >> selector: aSelector from: aClass [
 
-	^ (selectors
-		   inject: self trueCondition
-		   into: [ :cond :each |
-		   cond & (RBCondition definesSelector: each in: class) ])
-	 
+	^ self new
+		selector: aSelector from: aClass
 ]
 
 { #category : #preconditions }
 RBRemoveMethodRefactoring >> breakingChangePreconditions [
 
 	^ (RBCondition withBlock: [
-			   self checkSuperMethods.
-			   true ])
+		   self checkSuperMethods.
+		   true ]) & (RBCondition withBlock: self senders isEmpty errorString: 'Cannot remove method because it has senders')
```

Comment:
Shouldn't this be a block? If it is not it means that it is executed at the moment of the execution of this method, and not when we are sending #value to the block of RBCondition.

```suggestion
		   true ]) & (RBCondition withBlock: [ self senders isEmpty ] errorString: 'Cannot remove method because it has senders')
```